src/device/indexing.jl

```diff
@@ -65,7 +65,7 @@ macro linearidx(A, grididx=1, ctxsym=:ctx)
         x = $(esc(A))
         i = linear_index($(esc(ctxsym)), $(esc(grididx)))
         i > length(x) && return
-        i
+        i + firstindex(x) - 1
```

Comment:
Unrelated change that doesn't belong here? GPUArrays kernels aren't compatible with arbitrary array start indices, so there will be lots of breakage if you currently try this.

Reply:
I tried to use it to make `OffsetArray` on gpu broadcastable. https://github.com/JuliaGPU/Adapt.jl/pull/50 has been closed, so [3fd8c29](https://github.com/JuliaGPU/GPUArrays.jl/pull/393/commits/3fd8c29d35eea6fe960dd6b0ccf3b3e7cb48bdab) and [ded0eaa](https://github.com/JuliaGPU/GPUArrays.jl/pull/393/commits/ded0eaa831b06075b1e1f9dde6fc2e3096ed6ae6) will be removed.